shark/shark_runner_tf.py

```diff
@@ -0,0 +1,151 @@
+
+    def forward(self, input):
+        return get_results(self.iree_compilation_module, input,
+                           self.iree_config)
+
+
+class SharkInferenceTF:
+    """TODO: Write the description"""
+
+    def __init__(
+        self,
+        model,
+        input: tuple,
+        dynamic: bool = False,
+        device: str = None,
+        jit_trace: bool = False,
+        from_aot: bool = False,
+        custom_inference_fn=None,
+    ):
+        self.model = model
+        self.input = input
+        self.from_aot = from_aot
+
+        self.device = device if device is not None else shark_args.device
+
+        if from_aot:
+            aot_module = AOTModule(model,
+                                   input,
+                                   custom_inference_fn=custom_inference_fn)
+            aot_module.generate_inference_graph()
+            self.model = aot_module.forward_graph
+            self.input = aot_module.forward_inputs
+
+        self.shark_runner = SharkRunnerTF(self.model, self.input, dynamic,
+                                        self.device, jit_trace, from_aot)
+
+    def benchmark_forward(self, inputs):
+        inputs = self.input if self.from_aot else inputs
+        input_list = [x.numpy() for x in inputs]
+        for i in range(shark_args.num_warmup_iterations):
+            self.shark_runner.forward(input_list)
+
+        for i in range(shark_args.num_iterations):
+            begin = time.time()
+            out = self.shark_runner.forward(input_list)
+            end = time.time()
+            print("Iteration " + str(i) + ": " + str(end - begin))
+            if i == shark_args.num_iterations - 1:
+                return out
+
+    def forward(self, inputs):
+        # TODO Capture weights and inputs in case of AOT, Also rework the
+        # forward pass.
+        inputs = self.input if self.from_aot else inputs
+        input_list = [x.numpy() for x in inputs]
+        return self.shark_runner.forward(input_list)
+
+
+class SharkTrainerTF:
+    """TODO: Write the description"""
+
+    def __init__(
+        self,
+        model,
+        input: tuple,
+        dynamic: bool = False,
+        device: str = None,
+        jit_trace: bool = False,
+        from_aot: bool = True,
+        custom_inference_fn=None,
+    ):
+        self.model = model
+        self.input = []
+        self.from_aot = from_aot
+
+        self.device = device if device is not None else shark_args.device
+
+        aot_module = MakeFxModule(model,
+                                  input,
+                                  custom_inference_fn=custom_inference_fn)
+        aot_module.generate_graph()
+        self.model = aot_module.backward_graph
+
+        self.weights = [
+            i[1] for i in sorted(dict(model.named_parameters()).items())
+        ]
+        for i in sorted(dict(model.named_buffers()).items()):
+            self.weights.append(i[1])
+
+        for i in input:
+            self.input.append(i)
+
```

Comment:
I think `model.named_parameters()` is a PyTorch API call and TensorFlow would have a different call. This should change.